### DISCUSSION TITLE
core-rke2 images and AuroraBoot

### DISCUSSION THREAD

**Discussion by audiato** [kairos-io/kairos#1793] [Q&A]
Looking at the instructions for [AuroraBoot](https://kairos.io/docs/installation/netboot/#use-auroraboot) and I can see a reference to an opensuse-leap image in the `--set` parameter.
After having a dig around the Kairos quay repo I noticed there's a [core-opensuse-leap-rke2](https://quay.io/repository/kairos/core-opensuse-leap-rke2) image which includes installing rke2.

If I want to netboot nodes with RKE2, is this more or less ready to go, (accompanied by `--cloud-config` etc)?
Where does [provider-rke2](https://github.com/kairos-io/provider-rke2) come into play?

**Comment by Itxaka** [comment]
Hi!

Provider-rke2 comes from the repo https://github.com/kairos-io/provider-rke2 which has container releases under https://github.com/kairos-io/provider-rke2/pkgs/container/provider-rke2

To use those as a base image (its kairos+rk2) you can just set the parameter `--set "container_image=ghcr.io/kairos-io/provider-rke2:v2.3.3"`

And in the cloud config you would then add the extra config needed for rk2:

```
cluster:
  cluster_token: randomstring
  control_plane_host: cluster.example.com
  role: init
  config: |
    node-name: example-node
```

the main Auroraboot page in the docs has all the options possible to use when creating isos with aurora: https://kairos.io/docs/reference/auroraboot/

**Reply by audiato** [reply]
So that'll give me a full-fledged Kairos install with RKE2 instead of the standard K3s? Sweet! Thank you.

**Reply by audiato** [reply]
I'm getting the below error with this. If i run it with the following command:
`docker run -ti --net host quay.io/kairos/auroraboot --set "container_image=ghcr.io/kairos-io/provider-rke2:v2.3.3"`
or if I specify the config.yaml
`docker run -v "$PWD"/config.yaml:/config.yaml -ti --net host quay.io/kairos/auroraboot --set "container_image=ghcr.io/kairos-io/provider-rke2:v2.3.3" --cloud-config /config.yaml`

I get the below error.

```
1:11AM ERR Failed pulling container image 'ghcr.io/kairos-io/provider-rke2:v2.3.3' to '/tmp/temp-rootfs' (local: false): exit status 1
5 errors occurred:
	* exit status 1
	* 'gen-iso' deps container-pull failed
	* 'start-httpserver' deps gen-iso failed
	* 'extract-netboot' deps container-pull failed
	* 'start-netboot' deps extract-netboot failed
```

Have also tried other tags with the same result.
